MaintenanceEventsApi.yaml

```diff
@@ -0,0 +1,396 @@
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+        401:
+          description: Not authenticated
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+        403:
+          description: Forbidden
+          content: 
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+        404:
+          description: Could not find a target event uuid in the backend model
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+        406:
+          description: Not acceptable - only responds in application/json format
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+        500:
+          description: Server error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+      security:
+      - MaintenanceEventDefinitionAuth:
+        - fraihmwork.maintenance.write
+                
+
+components:
+  schemas:
+    Response:
+      type: object
+      description: A default uuid / message combination to convey simple success / error messages to the client
+      properties:
+        itemId:
+          type: string
+          format: uuid
+        message:
+          type: string
+          description: The success message provided back to the client
+      required:
+        - message
+    EventState:
+      type: string
+      enum:
+        - SCHEDULED
+        - CANCELED
+        - STARTED
+        - COMPLETED        
+    MaintenanceEventDefinition:
+      type: object
+      properties:
+        eventUuid:
+          type: string
+          format: uuid
+          description: The internal uuid of the maintenance event definition
+        name:
+          type: string
+          description: The name of the maintenance event          
+        description:
+          type: string
+          description: The description of the maintenance event
+        userContactUuid:
+          type: string
+          format: uuid
+          description: The User Contact uuid for who is responsible for performing the maintenance event
+        startTime:
+          type: string
+          format: date-time
+          description: Start time of maintenance event via ISO 8601 date-time format
+          example: "2021-01-30T08:30:00Z"
+        endTime:
+          type: string
+          format: date-time
+          description: End time of maintenance event via ISO 8601 date-time format
+          example: "2021-01-30T08:30:00Z"
+        componentUuid:
+          type: string
+          format: uuid
+          description: The uuid of the component scheduled for maintenance
+        eventState:
+          type: EventState
```

Comment:
eventState:
          $ref: '#/components/schemas/EventState'

We need to reference the schema here.
The description can be added within the EventState schema itself.

Reply:
Updated.